src/models/view/loginView.ts

```diff
@@ -31,11 +31,11 @@ export class LoginView implements View {
     }
 
     get canLaunch(): boolean {
-        return this.hasUris && this.uris[0].canLaunch;
+        return this.hasUris && this.uris.some(uri => uri.canLaunch);
     }
 
     get launchUri(): string {
-        return this.canLaunch ? this.uris[0].launchUri : null;
+        return this.hasUris ? this.uris.find(uri => uri.canLaunch).launchUri : null;
```

Comment:
Couldn't this lead to a null ref if find returns null here? seems you need to store find value in a variable and null check it

Reply:
Yes... :disappointed_relieved: